lib/engine/game/g_1848/step/buy_train.rb

```diff
@@ -8,6 +8,20 @@ module Game
     module G1848
       module Step
         class BuyTrain < Engine::Step::BuyTrain
+          def buy_train_action(action, entity = nil, borrow_from: nil)
+            entity ||= action.entity
+            price = action.price
+            remaining = price - buying_power(entity)
+
+            # do emergency loan
+            @game.perform_ebuy_loans(entity, remaining) if remaining.positive?
+
+            # company is closing, not buying train
+            return if entity.share_price.price.zero?
```

Comment:
If the company has the money to buy a train at this point (via the last loan that put them into receivership), do they buy the train before receivership?

Reply:
no, as per rules:
![image](https://user-images.githubusercontent.com/108803147/181685549-70abe74b-d0f4-40e1-90d3-6cd88e792cc1.png)

Reply:
(8.4.1) The company must buy a new train from the Bank and
may/must raise the money by taking out loans until either it has
enough cash to fund the purchase or its share price falls into the
“receivership” part of the Share Price Chart.